### ISSUE TITLE
EdgeShape.cubicCurve() returns QuadCurve, not CubicCurve

### ISSUE THREAD

**Issue by timboudreau** [jrtom/jung#69]
Looks like a copy/paste/forget-to-edit bug:

```
public static <V, E> EdgeShape<V, E>.QuadCurve cubicCurve(Graph<V, E> graph) {
        return new EdgeShape<V, E>(graph).new QuadCurve();
}
```

BTW, I'm not sure the refactoring of EdgeShape to make the curves inner, not nested, classes, is really a good idea - it adds overhead for the VM, and is completely avoidable here.  Also the outer instance of EdgeShape has to hold state and fields for shapes that will, in most cases, never be used.  I *think* you were trying to make the generic signature simpler - and it does that if you're writing Jung - not sure if that's the case if you're writing a [client of Jung](https://github.com/timboudreau/vl-jung/compare/jung-dev).

**Comment by jrtom** [comment]
Yup, thanks for catching the CubicCurve issue.  I've got a fix for that in progress.

I'll take another look at the static vs. inner aspect of the current design; filed #70 for this issue.

**Comment by jrtom** [comment]
FYI to those reading this thread, if you're not creating several different EdgeShape instances, you're better off using the static methods to create EdgeShape instances:

    EdgeShape<N, E> edgeShape = EdgeShape.line(graph);

etc., rather than creating a top-level EdgeShape and then creating a Line; the static method takes care of that for you.

(Reference to prior state of the code before the refactoring: http://grepcode.com/file/repo1.maven.org/maven2/net.sf.jung/jung-visualization/2.0.1/edu/uci/ics/jung/visualization/decorators/EdgeShape.java)

**Comment by jrtom** [comment]
Fixed in #71.
